beacon-chain/core/blocks/proposer_slashing_test.go

```diff
@@ -181,3 +184,73 @@ func TestProcessProposerSlashings_AppliesCorrectStatus(t *testing.T) {
 			newStateVals[1].ExitEpoch, beaconState.Validators()[1].ExitEpoch)
 	}
 }
+
+func TestVerifyProposerSlashing(t *testing.T) {
+	type args struct {
+		beaconState *stateTrie.BeaconState
+		slashing    *ethpb.ProposerSlashing
+	}
+
+	beaconState, _ := testutil.DeterministicGenesisState(t, 2)
+	currentSlot := uint64(0)
+	require.NoError(t, beaconState.SetSlot(currentSlot))
+
+	tests := []struct {
+		name    string
+		args    args
+		wantErr string
+	}{
+		{
+			name: "same header, no signatures",
+			args: args{
+				slashing: &ethpb.ProposerSlashing{
+					Header_1: &ethpb.SignedBeaconBlockHeader{
+						Header: &ethpb.BeaconBlockHeader{
+							ProposerIndex: 1,
+							Slot:          0,
+						},
+					},
+					Header_2: &ethpb.SignedBeaconBlockHeader{
+						Header: &ethpb.BeaconBlockHeader{
+							ProposerIndex: 1,
+							Slot:          0,
+						},
+					},
+				},
+				beaconState: beaconState,
+			},
+			wantErr: "expected slashing headers to differ",
+		},
+		{ // Regression test for https://github.com/sigp/beacon-fuzz/issues/74
+			name: "same header, different signatures",
+			args: args{
+				slashing: &ethpb.ProposerSlashing{
+					Header_1: &ethpb.SignedBeaconBlockHeader{
+						Header: &ethpb.BeaconBlockHeader{
+							ProposerIndex: 1,
+							Slot:          0,
+						},
+						Signature: bls.RandKey().Sign([]byte("foo")).Marshal(),
+					},
+					Header_2: &ethpb.SignedBeaconBlockHeader{
+						Header: &ethpb.BeaconBlockHeader{
+							ProposerIndex: 1,
+							Slot:          0,
+						},
+						Signature: bls.RandKey().Sign([]byte("bar")).Marshal(),
+					},
+				},
+				beaconState: beaconState,
+			},
+			wantErr: "expected slashing headers to differ",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			testutil.ResetCache()
+			if err := blocks.VerifyProposerSlashing(tt.args.beaconState, tt.args.slashing); (err != nil || tt.wantErr != "") && err.Error() != tt.wantErr {
```

Comment:
we should use our assertion library over here

Reply:
Does the assertion library support this behavior? 

I want: "if the test expects an error, assert the error is as expected"

Reply:
Not yet, it does have `assert.ErrorContains` which would be the closest to it.  But yeah maybe not worth it to change now.